packages/next-auth/src/providers/oauth.ts

```diff
@@ -17,7 +17,7 @@ type Client = InstanceType<Issuer["Client"]>
 
 export type { OAuthProviderType } from "./oauth-types"
 
-type ChecksType = "pkce" | "state" | "both" | "none"
+type ChecksType = "pkce" | "state" | "both" | "none" | "nonce"
```

Comment:
possibly should call this nonceValidation? to prevent people confusig it with `none`